addon/serializers/web-api.js

```diff
@@ -45,22 +48,32 @@ export default DS.RESTSerializer.extend({
     }
   },
 
-  extractErrors: function(store, typeClass, payload, id) {
-    let modelState = payload.modelState,
-        payloadKey = `${typeClass.modelName}.`,
-        formattedPayload = { errors: { } },
-        key;
+  normalizeErrors: function normalizeErrors(typeClass, payload) {
+    let payloadKey = `${typeClass.modelName}.`,
+        keys = Object.keys(payload);
 
-    for(key in modelState) {
-      if(modelState.hasOwnProperty(key)) {
-        formattedPayload.errors[key.replace(payloadKey, '').camelize()] = modelState[key];
+    keys.forEach(function(key) {
+      if(payload.hasOwnProperty(key)) {
+        payload[key.replace(payloadKey, '').camelize()] = payload[key];
+        delete payload[key];
       }
-    }
-
-    return this._super(store, typeClass, formattedPayload, id);
+    });
+    return this._super(typeClass, payload);
+  },
+  clearModelName: function(errors, modelName) {
```

Comment:
After merging the PR, I was looking through the code and trying to understand what this method does. Unfortunately I don't see any usages in this file and I also don't see that method in the `RESTSerializer` code (or the rest of the Ember Data repository for that matter).

Is this something you put in for your own use or am I missing something?